src/mixins/object_interactivity.mixin.js

```diff
@@ -231,28 +224,17 @@
     drawControls: function(ctx, styleOverride) {
       styleOverride = styleOverride || {};
       ctx.save();
-      var retinaScaling = this.canvas.getRetinaScaling(), matrix, p;
+      var retinaScaling = this.canvas.getRetinaScaling(), p;
       ctx.setTransform(retinaScaling, 0, 0, retinaScaling, 0, 0);
       ctx.strokeStyle = ctx.fillStyle = styleOverride.cornerColor || this.cornerColor;
       if (!this.transparentCorners) {
         ctx.strokeStyle = styleOverride.cornerStrokeColor || this.cornerStrokeColor;
       }
       this._setLineDash(ctx, styleOverride.cornerDashArray || this.cornerDashArray);
       this.setCoords();
-      if (this.group) {
-        // fabricJS does not really support drawing controls inside groups,
-        // this piece of code here helps having at least the control in places.
-        // If an application needs to show some objects as selected because of some UI state
-        // can still call Object._renderControls() on any object they desire, independently of groups.
-        // using no padding, circular controls and hiding the rotating cursor is higly suggested,
-        matrix = this.group.calcTransformMatrix();
```

Comment:
This hack was here because oCoords were wrong, but now they are fixed.
Why are they called oCoords????

Reply:
they were called oCoords when i joined.
Is not that oCoords were wrong, is that those coords are supposed to be coordinate relative to the html element, independetly of anything else going on in fabric, so that they could match the mouse cursor easily.

This hack was here to have controls rendered correctly for people that wanted to show a selection status in a group.

Reply:
Sorry i miss a piece of explanation.
Before oCoords where related to controls mouse interactions and nothing else, drawing code was static and independent.
If i m not confused, now oCoords contain the point where the controls will be drawn.
So now they take care of interaction and drawing, that all in all is more correct, at least you know that where is it drawn it will be also interactive.

Reply:
The hack is not needed anymore